lessons/A-Foundations-and-Tooling/01-windows-environment-setup.md

```diff
@@ -0,0 +1,64 @@
+# Lesson 1: Windows Environment Setup and VS Code Extensions
+
+This lesson guides you through setting up a professional development environment on Windows for modern Angular applications. We'll install Node.js, Yarn, and the Angular CLI, then configure Visual Studio Code with essential extensions for productivity and code quality.
+
+## 1. Install Core Dependencies
+
+A modern web development workflow starts with the right tooling. We'll use the official installers for each tool to ensure they are correctly set up and added to your system's PATH.
+
+### Node.js (LTS)
+
+Node.js is the JavaScript runtime that powers our development server, build tools, and dependency management. Always use the Long-Term Support (LTS) version for the best stability.
+
+- **Action:** [Download and install Node.js LTS](https://nodejs.org/en/download/) for Windows.
+- **Verification:** Open a new PowerShell or Command Prompt window and run `node -v` and `npm -v`. You should see the installed versions.
+
+### Yarn
+
+Yarn is a fast, reliable, and secure dependency manager. While `npm` is bundled with Node.js, Yarn offers improvements in performance and consistency.
+
+- **Action:** [Install Yarn](https://classic.yarnpkg.com/en/docs/install#windows-stable) using the recommended method for your system. The simplest is often via `npm`:
+  ```bash
+  npm install --global yarn
+  ```
+- **Verification:** In a new terminal, run `yarn --version`.
+
+### Angular CLI
+
+The Angular CLI is the official command-line interface for Angular. It helps you create, build, test, and deploy Angular applications.
+
+- **Action:** Install the Angular CLI globally using `npm`:
+  ```bash
+  npm install -g @angular/cli
+  ```
+- **Verification:** In a new terminal, run `ng --version`. You should see the Angular CLI banner and version details.
+
+## 2. Configure Visual Studio Code
+
+Visual Studio Code is our recommended code editor. Its rich ecosystem of extensions makes it a powerful tool for Angular development.
+
+- **Action:** [Download and install VS Code](https://code.visualstudio.com/download) if you haven't already.
+
+### Essential VS Code Extensions
```

Comment:
Every visual studio extension could have its own small lesson to show why it is useful

Reply:
i would prefer separate lesson for each. Just after this one.